File: packages/sdk-dart/lib/src/resources/usage.dart
import '../models/billing.dart';
import '../page.dart';
import '../transport.dart';

/// Usage: every billable line, and the totals.
class Usage {
  /// Binds the resource to a transport.
  const Usage(this._transport);

  final Transport _transport;

  /// Itemised events. Filter by [sessionId] to see exactly what one call
  /// cost; the sum equals what was debited.
  Future<Page<UsageEvent>> events({
    int? limit,
    String? startingAfter,
    String? endingBefore,
    String? sessionId,
    UsageComponent? component,
    DateTime? start,
    DateTime? end,
  }) =>
      _transport.list('/v1/usage/events', UsageEvent.fromJson, query: {
        'limit': limit,
        'starting_after': startingAfter,
        'ending_before': endingBefore,
        'session_id': sessionId,
        'component': component?.value,
        'start': start,
        'end': end,
      });

  /// Spend over a period, by component, provider and model.
  Future<UsageSummary> summary({DateTime? start, DateTime? end}) =>
      _transport.request(
        'GET',
        '/v1/usage/summary',
        UsageSummary.fromJson,
        query: {'start': start, 'end': end},
      );
}

File: packages/sdk-ts/src/resources/usage.ts
import type { Http, RequestOptions } from '../core/http.js';
import type { Page } from '../core/page.js';
import type { UsageEvent, UsageEventsQuery, UsageSummary, UsageSummaryQuery } from '../types.js';

/** Usage: every billable line, and the totals. */
export class Usage {
    constructor(private readonly http: Http) {}

    /** Itemised events. Filter by `session_id` to see exactly what one call cost; the sum equals what was debited. */
    events(query: UsageEventsQuery = {}, opts?: RequestOptions): Promise<Page<UsageEvent>> {
        return this.http.list<UsageEvent>('/v1/usage/events', query, opts);
    }

    /** Spend over a period by component, provider and model. Micro-USD: 1,000,000 = $1.00. */
    summary(query: UsageSummaryQuery = {}, opts?: RequestOptions): Promise<UsageSummary> {
        return this.http.request<UsageSummary>('GET', '/v1/usage/summary', { ...opts, query });
    }
}
